### ISSUE TITLE
Error in mice.impute.2lonly.pmm: object 'state' not found.

### ISSUE THREAD

**Issue by Niek-F** [stefvanbuuren/mice#97]
Hi,

When using "2lonly.pmm" as imputation method for a variable, mice returns the error: _Error in get("state", parent.frame(frame)) : object 'state' not found_. The error seems to be caused by a variable that is constant for certain levels of another variable. The use of any other method (for example pmm) does not result in an error. Somehow mice does not create a logged event when method is "2lonly.pmm", but I cannot figure out why. This is in mice version 3.0.0 and R version 3.5.0.
Any ideas that may help resolve this issue are much appreciated. 

I have made a small reproducible example that results in the error using the dataset from the mice.impute.2lonly.pmm help page:

```
G <- 250            # number of groups
n <- 20             # number of persons
beta <- .3          # regression coefficient
rho <- .30          # residual intraclass correlation
rho.miss <- .10     # correlation with missing response
missrate <- .50     # missing proportion
y1 <- rep( rnorm( G , sd = sqrt( rho ) ) , each=n ) + rnorm(G*n , sd = sqrt( 1 - rho )) 
w <- rep( round( rnorm(G ) , 2 ) , each=n )
v <- rep( round( runif( G , 0 , 3 ) ) , each=n )
x <-  rnorm( G*n ) 
y <- y1 + beta  * x + .2 * w + .1 * v
dfr0 <- dfr <- data.frame( "group" = rep(1:G , each=n ) , "x" = x , "y" = y , "w" = w , "v" = v )
dfr[ rho.miss * x + rnorm( G*n , sd = sqrt( 1 - rho.miss ) ) < qnorm( missrate ) , "y" ] <- NA
dfr[ rep( rnorm(G) , each=n ) < qnorm( missrate ) , "w" ] <- NA
dfr[ rep( rnorm(G) , each=n ) < qnorm( missrate ) , "v" ] <- NA

# Variables such as these cause the issue
dfr$test <- rep(1:20,length(unique(dfr$group)))
dfr$version[dfr$test == 1]<- sample(0:2,length(unique(dfr$group)),replace = T)
dfr$version[dfr$test > 1]<- 3

#....
# empty mice imputation
imp0 <- mice(dfr  , maxit=0 )
predM <- imp0$predictorMatrix
impM <- imp0$method

#...
# multilevel imputation
predM[c("y","v"),"group"] <- -2
predM["y","x"] <- 1        # fixed x effects imputation
impM[c("y","w","v")] <- c("2l.pan" , "pmm" , "2lonly.pmm" )


# y ... imputation using pan
# w ... imputation at level 2 using norm
# v ... imputation at level 2 using pmm

imp <- mice(dfr, m = 1, pred = predM , 
            method= impM, maxit = 1)
```

**Comment by gerkovink** [comment]
Hi @Niek-F, 

You have created a computationally singular system. Mice should fall back to ridge regression (`ls.meth = "ridge") from the default QR decomposition, but does not appear to do so for meth = "2lonly.pmm". 

Thanks for letting me know; I will fix this. In the meanwhile, you can run your code without errors by 

```
imp <- mice(dfr, m = 1, pred = predM , 
            method= impM, maxit = 1, ls.meth = "ridge")
```

All the best, 

Gerko

**Comment by stefvanbuuren** [comment]
Likely to be related to [this issue](https://github.com/alexanderrobitzsch/miceadds/pull/8).

**Comment by gerkovink** [comment]
The issue is related to a difference in parent frames. I have identified the issue and will fix the broader class of potential future issues.

**Comment by Niek-F** [comment]
ls.meth = "ridge" works and gives reasonable results. I've retried using version 3.0.8. but get a different error: Error in get("printFlag", parent.frame(4)) : object 'printFlag' not found
Thanks for all the help!

**Comment by gerkovink** [comment]
On second thought; setting the argument `print = FALSE` in `mice` is more elegant than defaulting to ridge regression altogether, because this leaves the parameter estimation default to QR decomposition for the other variables where no error is experienced. 

The error you experience is related to printing, not to estimation and since ridge regression introduces a small error only when needed, this may be avoided in cases where it is not needed. 

This issue is fixed in #98.

**Comment by stefvanbuuren** [comment]
I believe it's solved now, so I close
